src/extension/chatSessions/vscode-node/copilotCloudSessionsProvider.ts

```diff
@@ -173,25 +174,77 @@ export class CopilotCloudSessionsProvider extends Disposable implements vscode.C
 		@IInstantiationService instantiationService: IInstantiationService,
 		@IGithubRepositoryService private readonly _githubRepositoryService: IGithubRepositoryService,
 		@IChatDelegationSummaryService private readonly _chatDelegationSummaryService: IChatDelegationSummaryService,
+		@IExperimentationService private readonly _experimentationService: IExperimentationService,
 	) {
 		super();
-		const interval = setInterval(async () => {
-			const repoId = await getRepoId(this._gitService);
-			// TODO: handle no auth token case more gracefully
-			if (!this._authenticationService.permissiveGitHubSession) {
-				return;
-			}
-			const sessions = await this._octoKitService.getAllOpenSessions(repoId ? `${repoId.org}/${repoId.repo}` : undefined);
-			if (this.cachedSessionsSize !== sessions.length) {
-				this.refresh();
+
+		// Background refresh
+		getRepoId(this._gitService).then(async repoId => {
+			const telemetryObj: {
+				intervalMs?: number;
+				hasHistoricalSessions?: boolean;
+				isEmptyWindow: boolean;
+			} = {
+				isEmptyWindow: !vscode.workspace.workspaceFolders || vscode.workspace.workspaceFolders.length === 0
+			};
+			if (repoId) {
+				try {
+					const sessions = await this._octoKitService.getAllSessions(`${repoId.org}/${repoId.repo}`, false);
+					const hasHistoricalSessions = sessions.length > 0;
+					const intervalMs = this.getRefreshIntervalTime(hasHistoricalSessions);
+					telemetryObj.intervalMs = intervalMs;
+					telemetryObj.hasHistoricalSessions = hasHistoricalSessions;
+					const schedulerCallback = async () => {
+						// TODO: handle no auth token case more gracefully
+						if (!this._authenticationService.permissiveGitHubSession) {
+							return;
+						}
+						const sessions = await this._octoKitService.getAllSessions(`${repoId.org}/${repoId.repo}`);
+						if (this.cachedSessionsSize !== sessions.length) {
+							this.refresh();
+						}
```

Comment:
Oh, after doing this we do still need to actually called `scheduer.schedule()` because it's not an interval automatically any more